torchaudio/_internal/module_utils.py

```diff
@@ -43,14 +43,34 @@ def deprecated(direction: str, version: Optional[str] = None):
     Args:
         direction: Migration steps to be given to users.
     """
-    def decorator(func):
-        @wraps(func)
-        def wrapped(*args, **kwargs):
-            message = (
-                f'{func.__module__}.{func.__name__} has been deprecated '
-                f'and will be removed from {"future" if version is None else version} release. '
-                f'{direction}')
-            warnings.warn(message, stacklevel=2)
-            return func(*args, **kwargs)
+    def decorator(obj):
+        if isinstance(obj, type):
+            # Update the class to emit a warning when used
+            class Wrapper(obj):
+                def __init__(self, *args, **kwargs):
+                    message = (
+                        f'{obj.__module__}.{obj.__name__} has been deprecated '
+                        f'and will be removed from '
+                        f'{"future" if version is None else version} release. '
+                        f'{direction}')
+                    warnings.warn(message, stacklevel=1)
+                    self._wrapped = wrapped
```

Comment:
Looks like `wrapperd` is not defined yet. Could you fix this?

`NameError: free variable 'wrapped' referenced before assignment in enclosing scope`

You can check the error here; https://app.circleci.com/pipelines/github/pytorch/audio/3543/workflows/2b475e36-74ce-4a39-afe1-b316e234886d/jobs/104175